core/src/avm2/activation.rs

```diff
@@ -2069,11 +2069,10 @@ impl<'a, 'gc> Activation<'a, 'gc> {
     }
 
     fn op_convert_o(&mut self) -> Result<FrameControl<'gc>, Error<'gc>> {
-        let value = self
-            .pop_stack()
-            .coerce_to_object(self)
-            .map_err(|_| "Cannot convert null or undefined to object")?;
-
+        let value = self.pop_stack();
+        if matches!(value, Value::Null | Value::Undefined) {
+            return Err(make_null_or_undefined_error(self, value, None));
```

Comment:
I might be incorrect, but is `make_null_or_undefined_error` the correct error? Shouldn't this one be error #1034?

Reply:
let me check, sec

Reply:
nope:
undefined -> convert_o:
```
TypeError: Error #1010: A term is undefined and has no properties.
	at global$init()
```
null -> convert_o:
```
TypeError: Error #1009: Cannot access a property or method of a null object reference.
	at global$init()
```

Reply:
It's confusing yeah, but avmplus indeed uses a direct equivalent of our `make_null_or_undefined_error` in the same spot